internal/actions/csr/svc.go

```diff
@@ -110,7 +110,16 @@ func (h *ApprovalManager) runAutoApproveForCastAINodes(ctx context.Context) {
 
 	log := h.log.WithField("RunAutoApprove", "auto-approve-csr")
 	c := make(chan *Certificate, 1)
-	go WatchCastAINodeCSRs(ctx, log, h.clientset, c)
+	go func() {
+		for {
+			if err := WatchCastAINodeCSRs(ctx, log, h.clientset, c); err != nil {
```

Comment:
why infinite loop here? If watch broke informer will handle reconnect. Did you think of other reasons?

Reply:
You are right, the informers handle the reconnect logic, but the `createInformer` and `informer.AddEventHandler` might fail. In such case we would end up with a running cluster-controller, but without the auto approval working.

If the context is done, then no error is returned and this goroutine quits.

Reply:
Can we initialize informer in main func and pass it here? Because if informer can't be created then there is something wrong

Reply:
Should be possible, sounds like a good idea